### PR TITLE
Community: Allow adding ARNs as model_id to support Amazon Bedrock custom models

### PR DESCRIPTION
**Pull request by supreetkt**
- **Description:** Adds an additional class variable to `BedrockBase` called `provider` that allows sending a model provider such as amazon, cohere, ai21, etc. 
  Up until now, the model provider is extracted from the `model_id` using the first part before the `.`, such as `amazon` for `amazon.titan-text-express-v1` (see [supported list of Bedrock model IDs here](https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids-arns.html)). But for custom Bedrock models where the ARN of the provisioned throughput must be supplied, the `model_id` is like `arn:aws:bedrock:...` so the `model_id` cannot be extracted from this. A model `provider` is required by the LangChain Bedrock class to perform model-based processing. To allow the same processing to be performed for custom-models of a specific base model type, passing this `provider` argument can help solve the issues.
The alternative considered here was the use of `provider.arn:aws:bedrock:...` which then requires ARN to be extracted and passed separately when invoking the model. The proposed solution here is simpler and also does not cause issues for current models already using the Bedrock class.
  - **Issue:** N/A
  - **Dependencies:** N/A

### CODE REVIEW

File: libs/community/langchain_community/llms/bedrock.py [hunk 1]
```diff
@@ -229,9 +229,16 @@ class BedrockBase(BaseModel, ABC):
     config: Optional[Config] = None
     """An optional botocore.config.Config instance to pass to the client."""
 
+    provider: Optional[str]
```
Comment (3coins) [inline libs/community/langchain_community/llms/bedrock.py:232]:
```suggestion
    provider: Optional[str] = None
```

File: libs/community/langchain_community/llms/bedrock.py [hunk 2]
```diff
@@ -229,9 +229,16 @@ class BedrockBase(BaseModel, ABC):
...
+    """The model provider, e.g., amazon, cohere, ai21, etc. When not supplied, provider
+    is extracted from the first part of the model_id e.g. 'amazon' in 
+    'amazon.titan-text-express-v1'. Model providers for custom models can be specified
+    here with the ARN supplied as the model_id."""
```
Comment (3coins) [inline libs/community/langchain_community/llms/bedrock.py:236]:
```suggestion
    """The model provider, e.g., amazon, cohere, ai21, etc. When not supplied, provider
    is extracted from the first part of the model_id e.g. 'amazon' in 
    'amazon.titan-text-express-v1'. This value should be provided for model ids that does not
    have the provider in them, e.g., custom and provisioned models that have an ARN associated
    with them."""
```

File: libs/community/langchain_community/llms/bedrock.py [hunk 3]
```diff
@@ -353,7 +360,16 @@ def _identifying_params(self) -> Mapping[str, Any]:
...
+        if self.provider:
+            return self.provider
+        else:
+            if self.model_id.startswith("arn"):
+                raise ValueError(
+                    "Model provider should be supplied when passing a model ARN as "
+                    "model_id"
+                )
+            else:
+                return self.model_id.split(".")[0]
```
Comment (3coins) [inline libs/community/langchain_community/llms/bedrock.py:372]: else is extraneous here.

```suggestion
        if self.model_id.startswith("arn"):
            raise ValueError(
                "Model provider should be supplied when passing a model ARN as "
                "model_id"
            )
        else:
            return self.model_id.split(".")[0]
```

File: libs/community/langchain_community/llms/bedrock.py [hunk 4]
```diff
@@ -229,9 +229,16 @@ class BedrockBase(BaseModel, ABC):
...
+    """The model provider, e.g., amazon, cohere, ai21, etc. When not supplied, provider
+    is extracted from the first part of the model_id e.g. 'amazon' in 
+    'amazon.titan-text-express-v1'. Model providers for custom models can be specified
+    here with the ARN supplied as the model_id."""
+
     model_id: str
     """Id of the model to call, e.g., amazon.titan-text-express-v1, this is
-    equivalent to the modelId property in the list-foundation-models api"""
+    equivalent to the modelId property in the list-foundation-models api or it can be
+    used to supply the ARN of the fine-tuned model."""
```
Comment (3coins) [inline libs/community/langchain_community/llms/bedrock.py:241]:
```suggestion
    equivalent to the modelId property in the list-foundation-models api. For custom and
    provisioned models, an ARN value is expected."""
```

**Review by 3coins — COMMENTED** [review]
@supreetkt 
Thanks for working on these changes. Some minor comments on the docs, and suggestion to remove extraneous code. Looks good otherwise.

**Comment by 3coins** [comment]
@baskaryan 
Looks good, this should be ready to merge.

**Review by baskaryan — APPROVED** [review]

**Comment by baskaryan** [comment]
looks like there's a merge conflict, otherwise lgtm

**Comment by 3coins** [comment]
@baskaryan 
I need to take over this PR to resolve conflicts, seems like the last few commits have added unintended changes. Please hold off on merging yet. Thanks.

**Comment by 3coins** [comment]
@baskaryan 
All good now!

**Comment by supreetkt** [comment]
@baskaryan - please feel free to merge. The merge conflicts are now resolved. Thanks!

**Comment by baskaryan** [comment]
thank you @supreetkt!

**Pull request merged**